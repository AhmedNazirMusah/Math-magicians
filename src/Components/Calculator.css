.button-container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.button-container,
:nth-child(19),
:nth-child(16),
:nth-child(12),
:nth-child(8),
:nth-child(4) {
  background-color: rgb(255, 115, 0);
}

/* .button-container,
:nth-child(19),
:nth-child(16),
:nth-child(12),
:nth-child(8),
:nth-child(4).
 hover{
  background-color: rgba(11, 7, 5, 0.591);
} */

.display-div {
  display: flex;
  flex-direction: row-reverse;
  background: rgb(106, 125, 126);
  color: white;
  font-size: 40px;
  padding: 8px;
}

.button-container button {
  font-size: 40px;
  font-weight: 600;
  padding: 9px;
}

.button-container :nth-child(17) {
  grid-column: span 2;
}

.main-div {
  max-width: 70%;
  margin: 15% 10%;
}
