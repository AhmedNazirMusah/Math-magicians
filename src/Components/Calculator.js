import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [textContent, setTextContent] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const allbuttons = [
    'AC', '+/-', '%', 'รท',
    '7', '8', '9', 'x',
    '6', '5', '4', '-',
    '3', '2', '1', '+',
    '0', '.', '=',
  ];

  const handleClick = (e) => {
    const {
      target: { textContent: buttonName },
    } = e;
    const newState = calculate(textContent, buttonName);
    setTextContent(newState);
  };
  const { total, next } = textContent;
  return (
    <div className="main-div">
      <p>Let&apos;s do some math!</p>
      <div className="calculator-div">
        <div className="display-div">{next || total || 0 }</div>
        <section className="button-container">
          {allbuttons.map((btn) => (
            <button
              key={btn}
              name={btn}
              type="button"
              className="buttons"
              onClick={(e) => handleClick(e)}
            >
              {btn}
            </button>
          ))}
        </section>
      </div>
    </div>
  );
};

export default Calculator;
