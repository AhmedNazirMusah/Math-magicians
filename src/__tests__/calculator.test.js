import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../Components/Calculator';

it('Performs simple calculations', () => {
  const Calculate = render(<Calculator />);
  expect(Calculate).toMatchSnapshot();
});
