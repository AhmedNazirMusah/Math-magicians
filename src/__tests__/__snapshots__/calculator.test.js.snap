// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Performs simple calculations 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-div"
      >
        <p>
          Let's do some math!
        </p>
        <div
          class="calculator-div"
        >
          <div
            class="display-div"
          >
            0
          </div>
          <section
            class="button-container"
          >
            <button
              class="buttons"
              name="AC"
              type="button"
            >
              AC
            </button>
            <button
              class="buttons"
              name="+/-"
              type="button"
            >
              +/-
            </button>
            <button
              class="buttons"
              name="%"
              type="button"
            >
              %
            </button>
            <button
              class="buttons"
              name="รท"
              type="button"
            >
              รท
            </button>
            <button
              class="buttons"
              name="7"
              type="button"
            >
              7
            </button>
            <button
              class="buttons"
              name="8"
              type="button"
            >
              8
            </button>
            <button
              class="buttons"
              name="9"
              type="button"
            >
              9
            </button>
            <button
              class="buttons"
              name="x"
              type="button"
            >
              x
            </button>
            <button
              class="buttons"
              name="6"
              type="button"
            >
              6
            </button>
            <button
              class="buttons"
              name="5"
              type="button"
            >
              5
            </button>
            <button
              class="buttons"
              name="4"
              type="button"
            >
              4
            </button>
            <button
              class="buttons"
              name="-"
              type="button"
            >
              -
            </button>
            <button
              class="buttons"
              name="3"
              type="button"
            >
              3
            </button>
            <button
              class="buttons"
              name="2"
              type="button"
            >
              2
            </button>
            <button
              class="buttons"
              name="1"
              type="button"
            >
              1
            </button>
            <button
              class="buttons"
              name="+"
              type="button"
            >
              +
            </button>
            <button
              class="buttons"
              name="0"
              type="button"
            >
              0
            </button>
            <button
              class="buttons"
              name="."
              type="button"
            >
              .
            </button>
            <button
              class="buttons"
              name="="
              type="button"
            >
              =
            </button>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-div"
    >
      <p>
        Let's do some math!
      </p>
      <div
        class="calculator-div"
      >
        <div
          class="display-div"
        >
          0
        </div>
        <section
          class="button-container"
        >
          <button
            class="buttons"
            name="AC"
            type="button"
          >
            AC
          </button>
          <button
            class="buttons"
            name="+/-"
            type="button"
          >
            +/-
          </button>
          <button
            class="buttons"
            name="%"
            type="button"
          >
            %
          </button>
          <button
            class="buttons"
            name="รท"
            type="button"
          >
            รท
          </button>
          <button
            class="buttons"
            name="7"
            type="button"
          >
            7
          </button>
          <button
            class="buttons"
            name="8"
            type="button"
          >
            8
          </button>
          <button
            class="buttons"
            name="9"
            type="button"
          >
            9
          </button>
          <button
            class="buttons"
            name="x"
            type="button"
          >
            x
          </button>
          <button
            class="buttons"
            name="6"
            type="button"
          >
            6
          </button>
          <button
            class="buttons"
            name="5"
            type="button"
          >
            5
          </button>
          <button
            class="buttons"
            name="4"
            type="button"
          >
            4
          </button>
          <button
            class="buttons"
            name="-"
            type="button"
          >
            -
          </button>
          <button
            class="buttons"
            name="3"
            type="button"
          >
            3
          </button>
          <button
            class="buttons"
            name="2"
            type="button"
          >
            2
          </button>
          <button
            class="buttons"
            name="1"
            type="button"
          >
            1
          </button>
          <button
            class="buttons"
            name="+"
            type="button"
          >
            +
          </button>
          <button
            class="buttons"
            name="0"
            type="button"
          >
            0
          </button>
          <button
            class="buttons"
            name="."
            type="button"
          >
            .
          </button>
          <button
            class="buttons"
            name="="
            type="button"
          >
            =
          </button>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
